AndelaOSP/Andela-Socials
# #166153570: Event host should be able to connect the App to slack explicitly

#### What Does This PR Do?
- adds a modal that has instructions to connect to slack

#### Description Of Task To Be Completed
- As an event host, I should be able to explicitly connect the Andela Social app to Slack. After creating an event, if I am not already connected, then I should see a modal that prompts me to connect the Andela social app to Slack.

#### Any Background Context You Want To Provide?
- N/A

#### How can this be manually tested?
- set SLACK_AUTH_REDIRECT_URI in your environment variable to `${SECURE_FRONTEND_BASE_URL}/slacktokencallback`. P:S: This URL must be included as one of the Redirect URLs in the slack settings.
- on the frontend, create a new event.
- if a slack token isn't found on the db the response will trigger the slack modal to pop up
- if you click cancel, the modal disappears
- if you click on confirm, you are redirected to a slack auth page to confirm
- you will be redirected back to the frontend and a response toast message will appear

#### What are the relevant pivotal tracker stories?
[#166153570 Event host should be able to connect the App to slack explicitly](https://www.pivotaltracker.com/story/show/166153570)

#### Screenshot(s)
![image](https://user-images.githubusercontent.com/28973383/58793285-773a2f00-85fe-11e9-82d1-255ee8da6ee5.png)

![image](https://user-images.githubusercontent.com/28973383/59023642-e14e1080-8858-11e9-88f2-74f12228d191.png)
